src/Workspaces/Core/Portable/Storage/SQLite/Interop/SqlConnection.cs

```diff
@@ -179,12 +179,65 @@ public void RunInTransaction(Action action)
             }
         }
 
+        public TResult RunInTransaction<T, TResult>(Func<T, TResult> action, T userData)
```

Comment:
Don't like the fact that I copied everything, not sure how to refactor it to handle both scenarios. Maybe just rewrite the first one to by a `Func<T, TResult>` and return an empty valuetuple?